src/main/scala/mesosphere/marathon/core/appinfo/EnrichedTask.scala

```diff
@@ -16,9 +16,26 @@ case class EnrichedTask(
     reservation: Option[Reservation])
 
 object EnrichedTask {
-  def apply(instance: Instance, task: Task, healthCheckResults: Seq[Health],
-    servicePorts: Seq[Int] = Nil): EnrichedTask = {
-    new EnrichedTask(instance.runSpecId, task, instance.agentInfo, healthCheckResults, servicePorts = servicePorts,
-      reservation = instance.reservation)
+
+  object All {
```

Comment:
can we maybe find better names for these? I had hard time understanding that does EnrichedTask.App and EnrichedTask.All mean. Maybe EnrichedTask.Single and EnrichedTask.All ?

Or maybe the EnrichedTask name does not make sense anymore altogether, since it's a collection of tasks?

Maybe we can go with `EnrichedTasks` then?

Reply:
Heavy +1 for choosing better names.